src/logics/LIALogic.h

```diff
@@ -47,59 +51,51 @@ class LIALogic: public LALogic
-    bool        hasSortNum(PTRef tr) const override ;//{ return hasSortInt(getPterm(tr).symb()); }
+    bool        hasSortInt(SymRef sr) const { return sym_store[sr].rsort() == sort_INTEGER; }
 
-    PTRef       getTerm_NumZero() const override;// { return term_Int_ZERO; }
-    PTRef       getTerm_NumOne()  const override;// { return term_Int_ONE; }
-    PTRef       getTerm_NumMinusOne()  const override;// { return term_Int_ONE; }
+    PTRef       getTerm_NumZero() const override { return term_Int_ZERO; }
+    PTRef       getTerm_NumOne()  const override { return term_Int_ONE; }
+    PTRef       getTerm_NumMinusOne() const override { return term_Int_MINUSONE; }
 
-    virtual const SymRef get_sym_Num_TIMES () const override;
-    virtual const SymRef get_sym_Num_DIV () const override;
-    virtual const SymRef get_sym_Num_MINUS () const override;
-    virtual const SymRef get_sym_Num_PLUS () const override;
-    virtual const SymRef get_sym_Num_NEG () const override;
-    virtual const SymRef get_sym_Num_LEQ () const override;
-    virtual const SymRef get_sym_Num_GEQ () const override;
-    virtual const SymRef get_sym_Num_LT () const override;
-    virtual const SymRef get_sym_Num_GT () const override;
-    virtual const SymRef get_sym_Num_EQ () const override;
-    virtual const SymRef get_sym_Num_ZERO () const override;
-    virtual const SymRef get_sym_Num_ONE () const override;
-    virtual const SymRef get_sym_Num_ITE () const override;
-    virtual const SRef get_sort_NUM () const override;
+    virtual const SymRef get_sym_Num_TIMES () const override { return sym_Int_TIMES; }
+    virtual const SymRef get_sym_Num_DIV () const override { return sym_Int_DIV; }
+    virtual const SymRef get_sym_Num_MINUS () const override { return sym_Int_MINUS; }
+    virtual const SymRef get_sym_Num_PLUS () const override { return sym_Int_PLUS; }
+    virtual const SymRef get_sym_Num_NEG () const override { return sym_Int_NEG; }
+    virtual const SymRef get_sym_Num_LEQ () const override { return sym_Int_LEQ; }
+    virtual const SymRef get_sym_Num_GEQ () const override { return sym_Int_GEQ; }
+    virtual const SymRef get_sym_Num_LT () const override { return sym_Int_LT; }
+    virtual const SymRef get_sym_Num_GT () const override { return sym_Int_GT; }
+    virtual const SymRef get_sym_Num_EQ () const override { return sym_Int_EQ; }
+    virtual const SymRef get_sym_Num_ZERO () const override { return sym_Int_ZERO; }
+    virtual const SymRef get_sym_Num_ONE () const override { return sym_Int_ONE; }
+    virtual const SymRef get_sym_Num_ITE () const override { return sym_Int_ITE; }
+    const SymRef get_sym_Int_MOD() const { return sym_Int_MOD; }
+    const SymRef get_sym_Int_ABS() const { return sym_Int_ABS; }
+    virtual const SRef get_sort_Num () const override { return sort_INTEGER; }
+
+    const PTRef mkIntMod(PTRef first, PTRef second);
+    const PTRef mkIntAbs(PTRef arg);
+    PTRef mkNumDiv(vec<PTRef> const & args) override;
```

Comment:
Why this takes a `vec` and `mkIntMod` takes two `PTRef`s?

Reply:
I think this is how it was before. When it was common for LIA and LRA.

Should we distinguish between the real operator `/` and the integer operator `div` at the level of `mk` methods?

Reply:
This was already resolved.